server/app/repository/VersionRepository.java

```diff
@@ -95,7 +95,7 @@ public void publishNewSynchronizedVersion() {
               activeQuestion ->
                   !draftQuestionNames.contains(activeQuestion.getQuestionDefinition().getName()))
           // For each active question not associated with the draft, associate it with the draft.
-          .forEach(activeQuestionNotInDraft -> activeQuestionNotInDraft.addVersion(draft).save());
+          .forEach(activeQuestionNotInDraft -> draft.addQuestion(activeQuestionNotInDraft));
```

Comment:
Add a comment since the directionality is important.  Also do we not need to save?

Reply:
Whoops, forgot to add that. Replicated the comment when adding programs to the version. The M2M will be updated when saving the version. Fortunately, all of our test coverage confirms that the behavior is the same before and after :D.

Reply:
Ah I see!  The .save is kinda hidden here so I didn't see it either.